account.go

```diff
@@ -39,14 +39,11 @@ type AccountMeta struct {
 }
 
 func (a *AccountMeta) less(act *AccountMeta) bool {
-	if a.IsSigner && !act.IsSigner {
-		return true
-	} else if !a.IsSigner && act.IsSigner {
-		return false
+	if a.IsSigner != act.IsSigner {
+		return a.IsSigner
```

Comment:
| self (`a.IsIsigner`) | other (`act.IsSigner`) | Less |
|-|-|-|
|true|false|true|
|false|true|false|
|true|true|`<continue>`|
|false|false|`<continue>`|

Is there any details about the exact rules somewhere. Shouldn't the first two row be inverted?

Reply:
I cannot find any documents specifying the rules. But I think solana-web3.js should be a good reference: [Transaction.compileMessage](https://github.com/solana-labs/solana-web3.js/blob/f34d5de/src/transaction.ts#L205)

Reply:
I agree, I will write the full tests later on.